### PR TITLE
[modbus] Export thing handler interface for upcoming bundles

### PR DESCRIPTION
**Pull request by mrbig**
This is a follow up to PR #4053 

The modbus binding could easily be extended by other bundles to add device specific things and their handlers. There is already such a work in progress, see issue #3216 

Fortunately @ssalonen  did a nice work separating interfaces from actual implementations so exporting the base handler interfaces is quite straightforward.

This work is based on the zigbee and bluetooth package practices and work by @triller-telekom 

I've split the ModbusBindingConstants class also in two: one that is public and could be used by other bundles and one that remains internal.

### CODE REVIEW

File: addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/ModbusBindingConstants.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * The {@link ModbusBinding} class defines some constants
+ * public that might be used from other bundles as well.
+ *
+ * @author Sami Salonen - Initial contribution
+ * @author Nagy Attila Gabor - Split the original ModbusBindingConstants in two
+ */
+@NonNullByDefault
+public class ModbusBindingConstants {
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/ModbusBindingConstants.java]: Should we give this class (or the original [ModbusBindingConstants](https://github.com/openhab/openhab2-addons/pull/4077/files#diff-0af7815991dc3c4419a8c33927a1783e) class) a different name to make it clear for everyone that one of them is internal and the other one is public API? Wdyt?

Comment (mrbig) [reply addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/ModbusBindingConstants.java]: Totally agree, I just could not come up with a good name. As I can see this separation is fairly new, I would like to ask you guys to propose a naming convention here.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/ModbusBindingConstants.java]: Off-the-cuff: `ModbusBindingConstants` and `ModbusBindingConstantsInternal`. :wink:

Comment (mrbig) [reply addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/ModbusBindingConstants.java]: Ok, renamed the class

**Comment by mrbig** [comment]
Thank you!

**Pull request merged**